Source/missiles.cpp

```diff
@@ -570,23 +570,23 @@ void MoveMissileAndCheckMissileCol(Missile &missile, int mindam, int maxdam, boo
 	missile.lastCollisionTargetHash = tileTargetHash;
 }
 
-void SetMissAnim(Missile &missile, int animtype)
+void SetMissAnim(Missile &missile, MissileGraphicID animtype)
 {
 	int dir = missile._mimfnum;
 
-	if (animtype > MFILE_NONE) {
-		animtype = MFILE_NONE;
+	if (animtype > MissileGraphicID::None) {
+		animtype = MissileGraphicID::None;
 	}
 
 	missile._miAnimType = animtype;
-	missile._miAnimFlags = MissileSpriteData[animtype].flags;
+	missile._miAnimFlags = MissileSpriteData[static_cast<uint8_t>(animtype)].flags;
 	if (!HeadlessMode) {
-		missile._miAnimData = MissileSpriteData[animtype].spritesForDirection(static_cast<size_t>(dir));
+		missile._miAnimData = MissileSpriteData[static_cast<uint8_t>(animtype)].spritesForDirection(static_cast<size_t>(dir));
 	}
-	missile._miAnimDelay = MissileSpriteData[animtype].animDelay[dir];
-	missile._miAnimLen = MissileSpriteData[animtype].animLen[dir];
-	missile._miAnimWidth = MissileSpriteData[animtype].animWidth;
-	missile._miAnimWidth2 = MissileSpriteData[animtype].animWidth2;
+	missile._miAnimDelay = MissileSpriteData[static_cast<uint8_t>(animtype)].animDelay[dir];
+	missile._miAnimLen = MissileSpriteData[static_cast<uint8_t>(animtype)].animLen[dir];
+	missile._miAnimWidth = MissileSpriteData[static_cast<uint8_t>(animtype)].animWidth;
+	missile._miAnimWidth2 = MissileSpriteData[static_cast<uint8_t>(animtype)].animWidth2;

```

Comment:
Would be nice to just pull out `MissileSpriteData[static_cast<uint8_t>(animtype)]` in to it's own const value, and then the individual properties:
```cpp
const MissileData &missileData = MissileSpriteData[static_cast<uint8_t>(animtype)];

	missile._miAnimDelay = missileData.animDelay[dir];
	missile._miAnimLen = missileData.animLen[dir];
	missile._miAnimWidth = missileData.animWidth;
	missile._miAnimWidth2 = missileData.animWidth2;
```